src/main/java/org/truffleruby/core/array/ArrayUtils.java

```diff
@@ -222,6 +222,15 @@ public static int memcmp(final byte[] first, final int firstStart, final byte[]
         return 0;
     }
 
+    public static int memchr(byte[] ptr, int start, int find, int len) {
```

Comment:
Could you type `find` as `byte` here?
That would make the signature clearer (and potentially avoid extra sign extends).

Could you also rename `ptr` => `array` and `len` => `size` for consistency with other methods?